python2/koans/about_scoring_project.py

```diff
@@ -34,8 +34,21 @@
 # Your goal is to write the score method.
 
 def score(dice):
+    final_score = 0
+    repeated_nums = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+    for number in dice:
+        repeated_nums[number]=repeated_nums[number] + 1
+    for i in range(0,10):
```

Comment:
A couple comments:

Whenever you use a range and you want it to start from 0, you can skip the first parameter as it is implied.
```python
In [1]: range(10)
Out[1]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

In [2]: range(0, 10)
Out[2]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Speaking of ranges, in Python 2 there's two flavors of range: `range` and `xrange`. Both do exactly the same, but have different implementations that make `xrange` more efficient for loops. 

`range(x, y)` returns a list with elements that go from `x` to `y-1`. 

```python
In [1]: range(10)
Out[1]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```
which is good when you want to create such a list, but requires loading the entire list into memory. It's not really significant for small cases, but if you do `range(1000000)`, that will create a list and store one million `int`s in memory.

`xrange`, on the other hand, returns an ***iterator***. 

```python
In [1]: xrange(10)
Out[1]: xrange(10)
```

Iterators are one of Python's way of generating values as they are needed. Whenever you can ask an iterator to provide a new value (like in a `for` loop), it runs an internal function that determines the next number to return. In `xrange`'s case, it returns a number that increases or decreases by 1 each time a new value is requested, until it reaches the bound you set for it. The advantage here is that xrange only needs to load its internal function into memory, and can generate a million numbers with a smaller memory footprint. 

I recommend always using `xrange`, unless you specifically need a list containing all numbers. Python 3 replaced the `range` implementation with the `xrange` implementation, meaning there is no `xrange` and `range` always returns an iterator.


Finally, to make this `for` loop a little more generic, you can get the length of `repeated_nums` to set the bounds of your `for` loop.

```python
for i in xrange(len(repeated_nums))
```